libs/designsystem/modal/src/modal/services/modal.helper.ts

```diff
@@ -64,13 +65,24 @@ export class ModalHelper {
     }
 
     let canDismiss: boolean | (() => Promise<boolean>) = true;
+
+    if (config.canDismiss) {
+      canDismiss = this.canDismissHelper.getCanDismissCallback(config.canDismiss);
+    }
+
+    // This functionality is kept to prevent breaking changes, but should be depracated in the next major.
+    // It will be replaced by the new 'showAlert' callback.
     if (alertConfig) {
+      console.warn(
+        "This way of passing an alertConfig to the modal will be deprecated in the next major version. We recommend using the 'showAlert' callback instead."
+      );
+
       // Remembers the modal dismissal response from user to prevent multiple alerts on
       // approval since the callback is invoked more than once when closing.
```

Comment:
Not sure I understand why this is the case? Why would it be called more than once? 🤔

Reply:
I can't remember the specifics 100%, since it has to do with the old (and soon to be deprecated) implementation of showing an alert with a router based alert, and I didn't work on that implementation. But it is something along the lines of the `onWillDismiss` being called twice - one time from the `onWillDismiss` that is being returned when creating the modal, and the second time from the `hideAll` method in `modal.controller`, which is called when a router modal is closed.